### PR TITLE
[pipeline-connector][kafka] add kafka pipeline data sink connector.

### PR DESCRIPTION
**Pull request by lvyanquan**
This closes https://github.com/ververica/flink-cdc-connectors/issues/2691.
* support value format of `debeium-json` and `canal-json`.
* The written topic of Kafka will be `namespace.schemaName.tableName` string of TableId，this can be changed using `route` function of pipeline.
* If the written topic of Kafka is not existed, we will create one automatically.

### CODE REVIEW

**Comment by lvyanquan** [comment]
@Shawn-Hx PTAL.

File: docs/content/pipelines/kafka-pipeline(ZH).md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        <th class="text-left" style="width: 50%">Description</th>
+      </tr>
+    </thead>
+    <tbody>
+    <tr>
+      <td>type</td>
+      <td>required</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>指定要使用的连接器, 这里需要设置成 <code>'kafka'</code>.</td>
```
Comment (Shawn-Hx) [inline docs/content/pipelines/kafka-pipeline(ZH).md:55]: 中文逗号

File: docs/content/pipelines/kafka-pipeline(ZH).md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    </tr>
+    <tr>
+      <td>name</td>
+      <td>optional</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>Sink 的名称.</td>
```
Comment (Shawn-Hx) [inline docs/content/pipelines/kafka-pipeline(ZH).md:62]: 中文逗号

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka/pom.xml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        <groupId>com.ververica</groupId>
+        <version>${revision}</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>flink-cdc-pipeline-connector-kafka</artifactId>
+
+    <properties>
+        <zookeeper.version>3.5.9</zookeeper.version>
+        <testcontainers.version>1.17.2</testcontainers.version>
```
Comment (Shawn-Hx) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka/pom.xml:16]: Can this line be removed?

**Review by Shawn-Hx — COMMENTED** [review]
Thanks for the great contribution. Just left some small comments.

And can log4j2-test.properties be added?

**Comment by lvyanquan** [comment]
Thanks for these advice, addressed it.

**Review by Shawn-Hx — APPROVED** [review]
LGTM

File: docs/content/pipelines/kafka-pipeline(ZH).md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>Sink 的名称。 </td>
+    </tr>
+    <tr>
+      <td>value.format</td>
+      <td>optional</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>用于序列化 Kafka 消息的值部分数据的格式。</td>
```
Comment (ruanhang1993) [inline docs/content/pipelines/kafka-pipeline(ZH).md:69]: 我们固定支持了一些默认格式吗，可选的值是什么。

这一期用户是否可以自定义实现自己的写入格式，如果支持的话，怎么去操作。

**Comment by svea-vip** [comment]
I tested it, but did not get the output result I expected. The column names were changed to f1, f2, and I noticed that the schema change event was skipped in the code. I think it should be configurable here, and I am trying to compile and modify it. I hope to get your help.

**Comment by lvyanquan** [comment]
@svea-vip Hi，can you show more about your situation, and what's the output result you expected?
What impact will the change in table structure have if I only output data here.

**Comment by svea-vip** [comment]
@lvyanquan

**Comment by svea-vip** [comment]
@lvyanquan I would like to know your pipeline design for Kafka, which does not have independent metadata like a database. How do you plan to implement MetadataAccessor and MetadataApplier for Kafka.

**Comment by lvyanquan** [comment]
> @lvyanquan I would like to know your pipeline design for Kafka, which does not have independent metadata like a database. How do you plan to implement MetadataAccessor and MetadataApplier for Kafka.

As Kafka does not have independent metadata like a database, I actually do nothing in [MetadataApplier](https://github.com/ververica/flink-cdc-connectors/pull/2938/files#diff-0be6ff68652a3b78a179653e172f12908afda998aa2a0da609da1a5039e6cc13R81), and skip processing SchemaChangeEvent.

**Comment by lvyanquan** [comment]
> Specifies whether to write a topic, kakfka header record database & table name

I've added [two options](https://github.com/ververica/flink-cdc-connectors/pull/2938/files#diff-b817add666f310be030456dfae165e4b8c9f4111eac9522b6589564290118e1aR67-R80) to specify this, PTAL.

**Comment by lvyanquan** [comment]
> Can we use Schema Registry as Metadata provider ?

Yes, but I am still considering whether to output and how to output table structure changes.

**Comment by lvyanquan** [comment]
Rebased to master.

**Review by PatrickRen — COMMENTED** [review]
@lvyanquan Thanks for the PR! I left some comments

**Pull request merged**